lib/symboldatabase.cpp

```diff
@@ -6249,7 +6249,7 @@ static T* findTypeImpl(S& thisScope, const std::string & name)
     it = thisScope.definedTypesMap.find(emptyString);
     if (it != thisScope.definedTypesMap.end()) {
         for (S *scope : thisScope.nestedList) {
-            if (scope->className.empty() && (scope->type == thisScope.eNamespace || scope->isClassOrStructOrUnion())) {
+            if (scope->className.empty() && (scope->type == ScopeType::eNamespace || scope->isClassOrStructOrUnion())) {
```

Comment:
I filed https://github.com/llvm/llvm-project/issues/129591 about the `readability-static-accessed-through-instance` false negative.